validator/src/validator.rs

```diff
@@ -432,8 +477,11 @@ where
     }
 
     fn on_blockchain_history_adopted(&mut self, _: &Blake2bHash) {
-        self.mempool.mempool_clean_up();
-        debug!("Performed a mempool clean up because new history was adopted");
+        // Mempool updates are only done once we can be active.
+        if self.can_be_active() {
```

Comment:
Note on these:
As I understand it, the inconsistency of the mempool will subsequently fixed by #1475